### ISSUE TITLE
AutoSuspend Timeout issue

### ISSUE THREAD

**Issue by mergen3107** [koreader/koreader#7874]
* KOReader version: 2021-05.90
* Device: Kindle PW3

#### Issue
In KOReader we can set AutoSuspend timeout. If it is larger than 10 minutes, Kindle goes to sleep at 10 minutes because this is its default. 
Otherwise, it works fine (less than 10 minutes or Exit - Sleep). 

#### Steps to reproduce
Set AutoSuspend to 15 minutes. 
Wait. 
Kindle goes to sleep in 10 minutes. 

I suppose KOReader has no way to edit/override native sleep timer?

**Comment by mergen3107** [comment]
If AutoSuspend timeout is exactly 10 minutes, it works fine. I thought native suspend and KOReader suspend might overlap, but they didn’t. So I guess the issue is only about overriding the timeout.

**Comment by NiLuJe** [comment]
Isn't the plug-in disabled on Kindle?

**Comment by Frenzie** [comment]
Doesn't look like it. Maybe you meant to and forgot?

https://github.com/koreader/koreader/blob/ee92ca68e847f42fac140d6c5644b516eee930ff/plugins/autosuspend.koplugin/main.lua#L3-L11

**Comment by NiLuJe** [comment]
On mobile right now, but I vaguely recall that one / both of the options in there (suspend/shutdown) depend on a device cap that's false on Kindle...

**Comment by NiLuJe** [comment]
Huh., https://github.com/koreader/koreader/pull/7612

**Comment by mergen3107** [comment]
If I turn off the plugin, Kindle's default 10 min suspend still works.
So I tried to disable the screensaver time out in KUAL before launching KOReader and Kindle does not sleep at all (duh). 
So the problem is that the plugin cannot override the time out. 
Can this be fixed? 
Otherwise I'll just disable the plugin and use 10 min default. The plugin in this case is only useful for timeouts less than 10 min.

**Comment by NiLuJe** [comment]
Last I checked (roughly a dozen years ago xD), setting the t1 timeout at runtime was a PITA, so, unless someones has an idea, no, we should just disable the plugin again.

**Comment by mergen3107** [comment]
@NiLuJe OK! 
Thank you for your quick response :)

**Comment by NiLuJe** [comment]
Tangentially related to #9269

**Comment by melyux** [comment]
Any context on that linked PR @NiLuJe? Two new Kindle device states discovered? And are you going to still disable the plugin for Kindle? Seems sensible

**Comment by NiLuJe** [comment]
Nope, just had that tab open and wanted to keep the cross-ref ;o).

I still have zero interest in looking into this, so, yeah, disabling the plugin sounds good.

**Comment by NiLuJe** [comment]
... On the other hand, the reasoning behind #7612 still somewhat makes sense, if, for whatever reason the stock t1 timeout doesn't take.

TL;DR: Probably better to see if we can actually control the t1 timeout and keep the status quo in the mean time (as I'll take *more* suspend rather than less ;)).

**Comment by NiLuJe** [comment]
> TL;DR: Probably better to see if we can actually control the t1 timeout and keep the status quo in the mean time (as I'd rather take _more_ than less ;)).

Hmm, the fact that the Elektra db lives in a squashfs in the RO rootfs makes it doubly RO, meaning this would be a bitch to update at runtime without resorting to mount hackery that I do *not* want to deal with in KOReader :/.

**Comment by melyux** [comment]
Will that periodic refresh still apply when the plugin is disabled and/or affect the Kindle's energy saving in any way?

**Comment by NiLuJe** [comment]
Nope, it's tied to AutoSuspend being enabled and with a timeout set. At which point, it just delays the native t1 entry until it gets triggered by AutoSuspend.
